src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AuditApiAction.java

```diff
@@ -153,6 +160,21 @@ protected void handlePut(final RestChannel channel, final RestRequest request, f
         super.handlePut(channel, request, client, content);
     }
 
+    @Override
+    protected void handleGet(final RestChannel channel, RestRequest request, Client client, final JsonNode content) {
+        final SecurityDynamicConfiguration<?> configuration = load(getConfigName(), true);
+        filter(configuration);
+
+        final String resourcename = getResourceName();
+        if (!configuration.exists(resourcename)) {
+            notFound(channel, "Resource '" + resourcename + "' not found.");
+            return;
+        }
+
+        configuration.putCObject(READONLY_FIELD, readonlyFields);
```

Comment:
Can this by any chance override the existing list from the configuration?

Reply:
No. This will not override existing list in configuration. This is not stored in the index. Just added in the GET response. `load(getConfigName(), true)` will return a new object with configuration from index.

Reply:
Can it be part of the index? Why is it part of elasticsearch.yml?

Reply:
Mainly because of the following reasons:
- Configs need same types for keys. Cannot have different implementing classes for keys in a configuration. For eg: internalusers all the keys are of same implementing type.
- Admins also cannot change the readonly properties. Super-admin can bypass them. Not likely these settings are going to be changed often once deployed. 
- Considered storing in "_meta" of the document but we currently do expose APIs to update them. If at all user, user wants to change the readonly fields, then using super-admin we must replace the entire document after fetching the current configuration instead of updating in elasticsearch.yml (and doing a rolling restart). 

Let me know your thoughts

Reply:
Storing it in static_audit.yml resource file